iterable-extension/src/main/java/com/mparticle/ext/iterable/IterableExtension.java

```diff
@@ -65,7 +71,8 @@ private void processPushOpens(EventProcessingRequest processingRequest) throws I
                 if (event.getPayload() != null && processingRequest.getUserIdentities() != null) {
                     addUserIdentitiesToRequest(request, processingRequest);
                     if (request.email == null && request.userId == null) {
-                        throw new IOException("Unable to process PushMessageOpenEvent - user has no email or customer id.");
+                        logger.logError("Unable to process PushMessageOpenEvent - user has no email or customer id.");
+                        return;
```

Comment:
Based on the check in getPlaceholderEmail(), I don't think this scenario is possible. I think I'll just leave it to be safe though.